assets/js/app/Instance.js

```diff
@@ -18,46 +18,57 @@ LocusZoom.Instance = function(id, datasource, layout, state) {
     this.initialized = false;
 
     this.id = id;
-    this.parent = LocusZoom;
     
     this.svg = null;
 
-    // The _panels property stores child panel instances
-    this._panels = {};
+    // The panels property stores child panel instances
+    this.panels = {};
     this.remap_promises = [];
+
+    // The layout is a serializable object used to describe the composition of the instance
+    this.layout = layout || JSON.parse(JSON.stringify(LocusZoom.DefaultLayout));
```

Comment:
Ultimately layout values need to live somewhere. Many of the basic dimension-like layout values lived on an object on the Instance/Plot or the Panel called `view` which had *exactly* the same morphology as the layout object itself does. So in this branch we've dispensed with that and we are essentially using the layout to "inflate" each class (Instance/Plot, Panel, or DataLayer) but then we also cache what was used to do the inflation on the object for quick reference.

We actually haven't gotten too deep into the live editing of layouts yet... the only example of being able to do that now would be the resize UI. While I agree that we have to watch for property changes if the layout object stored on an Instance/Panel/DataLayer we'll have that problem no matter where we stash that information because all objects in JavaScript are editable from anywhere. So we may as well keep it simple and store them exactly how we consume them.

Finally, if we can build a Layout class that would provide an API for safely making live changes to a layout then we document that and hand-editing values in the layout cached on an object is the same as hand-editing values of any attribute of any object: not directly supported and likely to break things. This isn't a problem we should be too worried about... if someone wants to do that they're just gonna do that. If we give them methods for doing it safely most people will use those methods.

Also worth noting: all of this discussion is in reference to a general pattern we would like to have (layouts that can change on-the-fly from a variety of inputs) but realistically the primary use case of LocusZoom as we know it today and in the near future involves being able to define a highly customizable layout up front that *doesn't need to change in the course of a session*. That's what this branch aims to do, so all of this discussion of on-the-fly-editable layouts, while relevant, is not something that needs to be immediately addressed.